packit/local_project.py

```diff
@@ -398,12 +399,15 @@ def checkout_ref(self, ref: str):
 
     def checkout_pr(self, pr_id: Union[str, int]):
         """
-        Fetch selected PR and check it out. This will work for github and pagure, not for gitlab.
+        Fetch selected PR and check it out.
         """
+        is_gitlab = True if isinstance(self.git_service, GitlabService) else False
         logger.info(f"Checking out PR {pr_id}.")
         remote_name = self.remote or "origin"
         rem: git.Remote = self.git_repo.remotes[remote_name]
-        remote_ref = f"+refs/pull/{pr_id}/head"
+        remote_ref = "+refs/{}/{}/head".format(
+            "merge-requests" if is_gitlab else "pull", pr_id
```

Comment:
I would not change the old behaviour. (Some of us are used to it also from other tools like https://github.com/packit/upsint/ )

We can unintentionally break something with that change.